doc/analyzer_design.md

```diff
@@ -0,0 +1,53 @@
+# Design: Analyze the Machine Learning Mode in SQLFlow
+
+## Concept
+
+Although the machine learning model is widely used in many fields, it remains mostly a black box. [SHAP](https://github.com/slundberg/shap) is widely used by data scientists to explain the output of any machine learning model.
+
+This design doc introduces how to support the `Analyze SQL` in SQLFlow with SHAP as the backend and display the visualization image to the user.
+
+## User interface
+
+Users usually use a **TRAIN SQL** to train a model and then analyze the model using an **ANALYZE SQL**, the simple pipeline like:
+
+Train SQL:
+
+``` sql
+SELECT * FROM train_table
+TRAIN xgboost.Estimator
+WITH
+    train.objective = "reg:linear"
+COLUMN x
+LABEL y
+INTO my_model;
+```
+
+Analyze SQL:
+
+``` sql
+SELECT * FROM train_table
+ANALYZE my_model
+WITH
+  plots = force 
```

Comment:
Maybe add some comments about how to select the axis to plot like `shap.force_plot(explainer.expected_value, shap_values[0,:], X.iloc[0,:])`

Reply:
Will update this section in the next PR.